cmd/createorupdate/update.go

```diff
@@ -11,7 +12,19 @@ import (
 	"github.com/openshift/openshift-azure/pkg/log"
 )
 
-func update(ctx context.Context, cs *api.OpenShiftManagedCluster, p api.Plugin) error {
+func update(ctx context.Context, cs, oldCs *api.OpenShiftManagedCluster, p api.Plugin, azuredeploy []byte) error {
+	scaleOps := isScaleOp(cs, oldCs)
+	for _, scale := range scaleOps {
+		if scale.down {
+			// TODO: If we are scaling down, we need to drain first.
```

Comment:
We probably want to do this always, regardless of whether it's a scale op or a more complex update.
The kernel of the code is at https://github.com/openshift/openshift-azure/blob/master/cmd/createorupdate/update.go#L131 - just factor it out, loop the roles here and call it.

Reply:
Depends on whether we want to allow scale+upgrade in a single update.

Reply:
Btw how do we know that we are draining the correct VM that will get deleted during the ARM update? Is it always the highest index that gets removed first?

Reply:
This is not an issue after all, we just need to drain+delete the right number of VMs.